libs/SalesforceSDK/src/com/salesforce/androidsdk/config/LoginServerManager.java

```diff
@@ -199,6 +199,8 @@ public void reset() {
 		} catch (Exception e) {
 			Log.w("LoginServerManager.getLoginServersFromRuntimeConfig",
 					"Exception thrown while attempting to read array, attempting to read string value instead");
+		}
+		if (mdmLoginServers == null) {
```

Comment:
Turns out that if the hosts are not in an ```array```, an ```Exception``` is not thrown anymore :-( So, we need to attempt to parse a ```string``` outside the ```catch``` block for it to work.